lib/routes/password.js

```diff
@@ -36,6 +36,206 @@ module.exports = function (
   }
 
   var routes = [
+    {
+      method: 'POST',
+      path: '/password/change',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: {
+            authPW: isA.string().min(64).max(64).regex(HEX_STRING).required(),
+            wrapKb: isA.string().min(64).max(64).regex(HEX_STRING).required()
+          }
+        }
+      },
+      handler: function (request, reply) {
+        log.begin('Password.change', request)
+        let devicesToNotify, verifyHash, account, newKeyFetchToken, newSessionToken
+
+        const sessionToken = request.auth.credentials
+        const originatingDeviceId = sessionToken.deviceId
+        const { wrapKb, authPW}  = request.payload
+        const ip = request.app.clientAddress
+        const wantsKeys = requestHelper.wantsKeys(request)
+        const { uid } = sessionToken
```

Comment:
Somewhere in here, we need to check that last-auth time on the session and bail out if it was too long ago, potentially with a new errno for "stale authentication" or some-such.